src/types/ResponsePayload.ts

```diff
@@ -31,10 +31,27 @@ export type ResponsePayload = {
    */
   _id?: string;
 
+  /**
+   * Array of deprecation warnings (hidden if NODE_ENV=production)
+   */
+  deprecations?: {
+
+    /**
+     * Deprecation description
+     */
+    message: string;
+
+    /**
+     * Deprecated since this version
+     */
+    version: string;
+  }[];
```

Comment:
Prefer the more readable `Array<{...}>` notation
```suggestion
  deprecations?: Array>{

    /**
     * Deprecation description
     */
    message: string;

    /**
     * Deprecated since this version
     */
    version: string;
  }>;
```